### PR TITLE
Issue #154 - Feature/code refactoring

### PR DESCRIPTION
**Pull request by Juyoo**
### Your checklist for this pull request
🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your master!
- [x] Make sure you are making a pull request against the **dev** (left side). Also you should start *your branch* off *dev*.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.

### Description
Refactoring of the code :
- Add Mappers with MapStruct in for DTO/MODEL tranformations
- Usage of ResponseEntity inside controllers
- Move class in the proper package
- Usage of dependency injection by controller instead of using it by properties
- Usage of repositories only in the related service

Thank you for your contribution ! :tropical_drink:

### CODE REVIEW

File: backend/pom.xml [hunk 1]
```diff
@@ -288,6 +290,18 @@
...
+        <dependency>
+            <groupId>org.mapstruct</groupId>
+            <artifactId>mapstruct-jdk8</artifactId>
+            <version>${org.mapstruct.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.mapstruct</groupId>
+            <artifactId>mapstruct-processor</artifactId>
+            <version>${org.mapstruct.version}</version>
+            <scope>provided</scope>
```
Comment (yodamad) [inline backend/pom.xml]: Why provided ? Add comment to explain

File: backend/src/test/java/io/suricate/monitoring/service/scheduler/NashornWidgetSchedulerTest.java [hunk 2]
```diff
@@ -20,10 +20,10 @@
 //@SpringBootTest
```
Comment (yodamad) [inline backend/src/test/java/io/suricate/monitoring/service/scheduler/NashornWidgetSchedulerTest.java]: No commented code, plz

File: backend/pom.xml (hunk 1 above)
Comment (Juyoo) [reply backend/pom.xml]: You right this is not useful, i will delete it

**Comment by yodamad** [comment]
Unit test in failure : NashornWidgetExecuteAsyncTaskTest.testNetworkCall

**Comment by Juyoo** [comment]
I'm currently working on NashornWidgetExecuteAsyncTaskTest and i saw this failure, i'm trying to find out why this Unit test is in failure. It works when lanching it by intellij but not with : mvn test

**Comment by yodamad** [comment]
Don't know if i've to remove branch ?

**Comment by Juyoo** [comment]
You could, i've deleted it

**Pull request merged**